apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/TraceHeaderContext.cs

```diff
@@ -50,13 +50,16 @@ internal sealed class TraceHeaderContext
         /// <summary>Gets the span id or null if none is available.</summary>
         public ulong? SpanId { get; }
 
-        /// <summary>True if the request should be traced.</summary>
-        public bool ShouldTrace { get; }
+        /// <summary>
+        /// True if the request should be traced.
+        /// Null if the trace header did not indicate whether or not it should be traced.
```

Comment:
Null is the value being unset in the header.  True or false in a header should override everything. 

[traceid]/[spanid];o=1 (This should be traced)
[traceid]/[spanid];o=0 (This should not be traced)
[traceid]/[spanid];  (This can trace or not be traced.)